ocean/src/social.rs

```diff
@@ -1,10 +1,15 @@
+use std::collections::HashMap;
+
 #[derive(Debug)]
 pub struct SocialNetwork {
-    //TO-DO: Add appropriate field
+    friendships: HashMap<String, Vec<String>>
```

Comment:
How do you feel about using a `HashSet<String>` instead of `Vec<String>` for the values of the HashMap? We still wouldn't add the same friend twice, but we could be more concise by using `if !friends.insert(friend_id) { //return Err }` in `add_friend`.  Similarly, we could use `if friends.remove(&friend_id) { return Ok() }` in `remove_friend`.